### ISSUE TITLE
Regularizer defined inside `get_output`

### ISSUE THREAD

**Issue by EderSantana** [fchollet/keras#689]
I have a regularizer (the reparametrization trick for variational autoencoder for ex.) that I can only define inside the `get_output` method. I define it and add it to the `self.regularizers` list inside that method. My question is, will it still get compiled? Anybody have an idea of how can I test this?

**Comment by EderSantana** [comment]
I just figured it out. The answer is yes, we can. But we have to go back and add the regularizers manually before we compile the model. Like this:
```python
model = Sequential()
tl = TestLayer(10, 10)  # layer with regularizer defined inside `get_output`
model.add(tl)
model.get_output()  # generate regularizer
print model.regularizers  # prints []
model.regularizers = tl.regularizers
print model.regularizers  # prints [_regularizers_class_]
model.compile('adam', 'mse')
```

@fchollet do you know of a better way to do this? I mean, in a native Keras way? If not, please just close the issue. Tkx!

**Comment by fchollet** [comment]
You should probably be able to add your regularizers in a `layer.init_updates` method, like this one: https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L45

This method is called on the layer when it is added to a model, right after it is connected to its parent.

**Comment by EderSantana** [comment]
I see, init_updates sounds like the way to go. Thanks!!!

**Comment by EderSantana** [comment]
Yeah, it works, all that I have to do is call `get_output` inside `init_updates`. So clean, loved it!

**Comment by nitish11** [comment]
Hi,

I couldn't find layer.init_updates method in the latest code.

Could you let me know the equivalent method in the latest build??

**Comment by EderSantana** [comment]
look for 
```python
def build(self):
```

**Comment by nitish11** [comment]
@EderSantana  :+1:   Thanks for the reply 

But, could you explain a bit.  How model.get_output()  will generate regularizer?

https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L29

One more issue:

In reference to the models code, please check the link below.
https://github.com/fchollet/keras/blob/master/keras/models.py#L403
Where is the initialization of self.regularizers????


Like for optimizers, we have
  self.optimizer = optimizers.get(optimizer)
Ref: https://github.com/fchollet/keras/blob/master/keras/models.py#L365

**Comment by EderSantana** [comment]
don't worry about models. Check class `Dense` https://github.com/fchollet/keras/blob/master/keras/layers/core.py#L666-L689
